doc/site/docs/02-core-concepts/04-requests.md

```diff
@@ -0,0 +1,360 @@
+---
+sidebar_position: 4
+---
+
+# Requests
+
+The `Request` object represents an incoming HTTP request to your Relic server. It provides access to all information about the client's request, including the HTTP method, URL, headers, query parameters, and body content.
+
+Every handler in Relic receives a request through the context object (`ctx.request`), allowing you to inspect and process the incoming data before generating a response.
+
+## Understanding the Request Object
+
+When a client makes a request to your server, Relic creates a `Request` object that encapsulates all the details of that request. This object is immutable and provides type-safe access to request data.
+
+The request flows through your middleware pipeline and reaches your handler, where you can extract the information you need to process the request and generate an appropriate response.
+
+### Key Request Properties
+
+The `Request` object exposes several important properties:
+
+- **`method`** - The HTTP method (GET, POST, PUT, DELETE, etc.) as a `Method` enum value
+- **`url`** - The relative URL from the current handler's perspective, including query parameters
+- **`requestedUri`** - The complete original URI that was requested
+- **`headers`** - Type-safe access to HTTP headers
+- **`body`** - The request body as a readable stream
+- **`protocolVersion`** - The HTTP protocol version (typically "1.1")
+
+## Accessing Request Data
+
+### HTTP Method
+
+The request method indicates what action the client wants to perform. Relic uses a type-safe `Method` enum rather than strings.
+
+```dart
+router.get('/info', (ctx) {
+  final method = ctx.request.method;  // Method.get
+  
+  return ctx.respond(Response.ok(
+    body: Body.fromString('Received a ${method.name} request'),
+  ));
+});
+```
+
+Common methods include `Method.get`, `Method.post`, `Method.put`, `Method.delete`, `Method.patch`, and `Method.options`.
+
+### Request URL and Path
+
+The `url` property provides the relative path and query parameters from the current handler's perspective. This is useful when your handler is mounted at a specific path prefix.
+
+```dart
+router.get('/users/:id', (ctx) {
+  final url = ctx.request.url;
+  final fullUri = ctx.request.requestedUri;
+  
+  print('Relative URL: $url');
+  print('Full URI: $fullUri');
+  
+  return ctx.respond(Response.ok());
+});
+```
+
+For a request to `http://localhost:8080/users/123?details=true`:
+
+- `url.path` would be relative to the handler.
+- `requestedUri` would be the complete url, including the query parameters.
+
+## Working with Query Parameters
+
+Query parameters are key-value pairs appended to the URL after a question mark (`?`). They're commonly used to pass optional data or filters to your endpoints.
```

Comment:
The plan is to add typed interface for query and path parameters as well before 1.0. So be prepared for some changes here.

Reply:
You can resolve this comment when you have seen it.